GoogleDataTransport/GDTCORLibrary/GDTCOREventMetricsCounter.m

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#import <Foundation/Foundation.h>
+
+#import "GoogleDataTransport/GDTCORLibrary/Private/GDTCOREventMetricsCounter.h"
+
+#import "GoogleDataTransport/GDTCORLibrary/Public/GoogleDataTransport/GDTCOREvent.h"
+
+typedef NSDictionary<NSNumber *, NSNumber *> GDTCORDroppedEventCounter;
+
+@interface GDTCOREventMetricsCounter ()
+
+// TODO(ncooke3): Document.
+@property(nonatomic)
+    NSDictionary<NSString *, GDTCORDroppedEventCounter *> *droppedEventCounterByMappingID;
```

Comment:
Just a thought I wonder if you need a serial queue to guard the read/write operations for this property? Similar to remote config design.

Reply:
Here's an example: https://github.com/firebase/firebase-ios-sdk/pull/9807/files

Reply:
Good thought. I would say no because, in practice, an instance of this class shouldn't be passed into multiple concurrent contexts.

To actually enforce that though. There are several safeguards in place:
- This property consists of only immutable types. (And new `NSString` keys are copied when being used added to the dictionary.)
- This class has one instance method and the instance method does not change the contents of this property. In fact, it creates a *new* instance of this class with the modifications.

I experimented with a more functional approach to prevent unintended data changes.